stream-tests/src/test/java/org/apache/pekko/stream/javadsl/SourceTest.java

```diff
@@ -815,6 +816,19 @@ public void mustBeAbleToUseMapWithResource() {
     Assert.assertFalse(gate.get());
   }
 
+  @Test
+  public void mustBeAbleToUseMapWithAutoCloseableResource() {
+    final TestKit probe = new TestKit(system);
+    final AtomicInteger closed = new AtomicInteger();
+    Source.from(Arrays.asList("1", "2", "3"))
+        .mapWithResource(() -> (AutoCloseable) closed::incrementAndGet, (resource, elem) -> elem)
+        .runWith(Sink.foreach(elem -> probe.getRef().tell(elem, ActorRef.noSender())), system);
+
+    Assert.assertEquals(closed.get(), 0);
```

Comment:
This will be flaky, maybe we can just remove it.

Reply:
✅ updated